cli/download_deno_binary.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use std::path::Path;
+use std::path::PathBuf;
+
+use crate::http_util::HttpClient;
+use crate::http_util::HttpClientProvider;
+use crate::util::progress_bar::ProgressBar;
+use crate::util::progress_bar::ProgressBarStyle;
+use deno_core::anyhow::bail;
+use deno_core::error::AnyError;
+
+use crate::cache::DenoDir;
+use crate::shared::ReleaseChannel;
+
+pub async fn download_deno_binary(
+  http_client_provider: &HttpClientProvider,
+  deno_dir: &DenoDir,
+  binary_kind: &str,
+  target: &str,
```

Comment:
Consider making these two enums that have `FromStr` impl

Reply:
Wouldn't that be an overkill for this case? This function is called from 2 places and `FromStr` will have "deno" and "denort" hardcoded anyways

Reply:
It is, until someone does a typo and put `deni` or `denp` :P